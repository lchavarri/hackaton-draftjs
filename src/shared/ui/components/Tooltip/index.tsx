import React, { ReactElement } from 'react';
import { Popup } from 'semantic-ui-react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faExclamationTriangle,
  faInfoCircle
} from '@fortawesome/pro-light-svg-icons';

type Props = {
  trigger: ReactElement;
  text: string;
  severity: 'info' | 'warning' | 'danger';
  position?:
    | 'top left'
    | 'top right'
    | 'bottom right'
    | 'bottom left'
    | 'right center'
    | 'left center'
    | 'top center'
    | 'bottom center'
    | undefined;
  size?: 'small' | 'mini' | 'tiny' | 'large' | 'huge' | undefined;
};

export default function Tooltip(props: Props) {
  const { trigger, text, severity, position, size } = props;

  return (
    <Popup
      size={size || 'small'}
      wide="very"
      position={position || 'left center'}
      trigger={trigger}
    >
      {getIcon(severity)}
      &nbsp;<b>{getTitle(severity)}</b> {text}
    </Popup>
  );
}

const getTitle = (severity: string) => {
  switch (severity) {
    case 'warning':
      return 'Warning!';
    case 'danger':
      return 'Danger!';
    case 'info':
    default:
      return 'Information';
  }
};

const getIcon = (severity: string) => {
  switch (severity) {
    case 'warning':
      return (
        <FontAwesomeIcon
          color="orange"
          icon={faExclamationTriangle}
          aria-label="Warning"
        />
      );
    case 'danger':
      return (
        <FontAwesomeIcon
          color="red"
          icon={faExclamationTriangle}
          aria-label="Danger!"
        />
      );
    case 'info':
    default:
      return (
        <FontAwesomeIcon
          color="green"
          icon={faInfoCircle}
          aria-label="Information"
        />
      );
  }
};
