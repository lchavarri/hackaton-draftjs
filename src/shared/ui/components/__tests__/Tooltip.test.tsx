import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import Tooltip from '../Tooltip';

describe('Tooltip component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {};
  const getComponent = () => <Tooltip {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
