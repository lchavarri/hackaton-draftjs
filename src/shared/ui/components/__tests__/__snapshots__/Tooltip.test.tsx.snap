// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component should match component snapshot 1`] = `
<Popup
  disabled={false}
  offset={0}
  on="hover"
  pinned={false}
  position="left center"
  size="small"
  wide="very"
>
  <mockConstructor
    aria-label="Information"
    color="green"
    icon={[MockFunction]}
  />
  Â 
  <b>
    Information
  </b>
   
</Popup>
`;
